### ISSUE TITLE
Default tsconfig.json settings produces many errors

### ISSUE THREAD

**Issue by gregoryagu** [aurelia/cli#773]
**I'm submitting a bug report**

* **Library Version:**
0.31.3

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Node Version:**
7.7.1

* **NPM Version:**
3.10.10

TypeScript 2.5

Webpack

**Current behavior:**
37 errors are generated after creating a new aspnetcore 2.0 project and adding Aurelia. For example:

 `Severity	Code	Description	Project	File	Line	Suppression State
Error	TS5055	Build:Cannot write file 'C:/Dev/ServiceHistory/ServiceHistory.04/ServiceHistory/webpack.config.js' because it would overwrite input file.	ServiceHistory	C:\Dev\ServiceHistory\ServiceHistory.04\ServiceHistory\tsc	1	
`

These errors can be resolved by setting `"allowJs": true,` to `"allowJs": false,`

Also generated is this error:

`Severity	Code	Description	Project	File	Line	Suppression State
Error	TS5051	(TS) Option 'sourceRoot can only be used when either option '--inlineSourceMap' or option '--sourceMap' is provided.	ServiceHistory (tsconfig project)	C:\Dev\ServiceHistory\ServiceHistory.04\ServiceHistory\tsconfig.json	9	Active
`

This can be resolved by removing `"sourceRoot": "src",` from the config file.

* **What is the expected behavior?**
No compile errors.

**Comment by JeroenVinke** [comment]
@gregoryagu is this still an issue with the new aspnet core setup?

**Comment by gregoryagu** [comment]
Yes, I just a few hours ago download Ed the master branch and tested with
this result.

**Comment by gregoryagu** [comment]
@JeroenVinke To fix the typescript compile errors:

1. Remove the `"sourceRoot" : src` line. This setting is for telling the browser debugger where to find the debug files. At runtime, the browser never accesses the files in src, so this setting cannot be correct.  I think someone thinks it means where the *.ts files are located to be compiled, but this is not what it is for.

2. Remove '"allowJs": true'  This setting tells Typescript to compile javascript files. Typically they would be compiled and placed in an output directory.  But we are not using an "outDir" parameter, so they are compiled to the same location.  But a js file cannot be compiled and then output to the same location as it would overwrite the original file.  So this setting cannot be correct.

So if these two lines are removed, typescript will work just fine.

**Comment by gregoryagu** [comment]
@JeroenVinke Note that I did test the change to instructions.txt:

`<TypeScriptCompileBlocked>true</TypeScriptCompileBlocked>`

With this line, typescript no longer compiles in VS, or otherwise.  After building, the "dist" folder is not created and no bundles are created. So I would revert the changes to editing csproj.

**Comment by JeroenVinke** [comment]
@gregoryagu With these changes I had no other issues, so we should do some comparing. Visual Studio should not be transpiling typescript, it should leave that to webpack, hence the blocking.

Instead of adding Aurelia to an existing asp.net project (created with dotnet new) , it may be useful to try and have the aurelia cli generate the asp.net core bits, by running `au new` instead of `dotnet new` and `au new --here` and selecting webpack, asp.net core and typescript.

If you could zip up the project that you are having issues with and share it then I can have a look

**Comment by gregoryagu** [comment]
@JeroenVinke I tried as you suggested with using au new.  I then ran it with au run.  It did indeed compile the typescript just fine and show the app as expected.

However, I now try to open the project in Visual Studio, and run from there.  It does indeed build and run.  Then I enter some invalid code into a typescript file.  This should be caught by the editor as well as the compiler.  But since Typescript is blocked, the editor and compiler ignore the invalid code.  When I then run with au run, the typescript compiler now does find the invalid code, and the bundles are not built.  If I run from within VS, the home page is displayed with "Loading..."

So by blocking typescript in the project file, VS no longer reports invalid code and it is only caught in the command window when it builds.

To make it work, I remove the line about VS blocking typescript, and then remove the two tsconfig items I mention above.
